packages/vx-brush/src/BaseBrush.tsx

```diff
@@ -46,28 +47,46 @@ export type UpdateBrush =
   | ((prevState: Readonly<BaseBrushState>, props: Readonly<BaseBrushProps>) => BaseBrushState);
 
 export default class BaseBrush extends React.Component<BaseBrushProps, BaseBrushState> {
+  private constructor(props: BaseBrushProps) {
```

Comment:
I think you can remove the constructor and simplify this whole thing to this

```tsx
  state = {	
    start: { x: 0, y: 0 },	
    end: { x: 0, y: 0 },	
    ...this.props.initialBrushPosition,
    extent: this.props.initialBrushPosition ? 
      this.getExtent(
          this.props.scrubberInitialState.start,
          this.props.scrubberInitialState.end,
      ) : {	
      x0: -1,	
      x1: -1,	
      y0: -1,	
      y1: -1,	
    },	
    bounds: {	
      x0: 0,	
      x1: this.props.width,	
      y0: 0,	
      y1: this.props.height,	
    },	
    isBrushing: false,	
    activeHandle: null,	
  };
```

Reply:
We need the constructor in order to get around `this.getExtent` being used before its initialization, barring pasting the function about the state initialization, but that's a little unconventional. Thoughts?

Reply:
ah, I see. 

I think the `constructor` is fine, but could we update to the more succinct / immutable syntax I suggested here (assigning to `this.state` in the `constructor` instead)?